Sources/SwiftInspectorVisitors/FunctionDeclarationVisitor.swift

```diff
@@ -14,16 +17,41 @@ public final class FunctionDeclarationVisitor: SyntaxVisitor {
 }
 
 fileprivate final class FunctionSignatureVisitor: SyntaxVisitor {
+  override func visit(_ node: FunctionParameterSyntax) -> SyntaxVisitorContinueKind {
+    guard
+      let externalLabel = node.firstName?.text,
+      let type = node.type?.typeDescription
+    else { return .skipChildren }
+
+    appendArgument(externalLabel: externalLabel, type: type)
+    return .skipChildren
+  }
   override func visit(_ node: ReturnClauseSyntax) -> SyntaxVisitorContinueKind {
     returnType = node.returnType.typeDescription
     return .skipChildren
   }
 
+  fileprivate func appendArgument(externalLabel: String, type: TypeDescription) {
+    var arguments = self.arguments ?? []
+    arguments.append(.init(externalLabel: externalLabel, type: type))
+    self.arguments = arguments
+  }
+
+  fileprivate var arguments: [FunctionDeclarationInfo.ArgumentInfo]?
   fileprivate var returnType: TypeDescription?
 }
 
 public struct FunctionDeclarationInfo: Codable, Hashable {
   public let name: String
+  public let arguments: [ArgumentInfo]?
   public let returnType: TypeDescription?
-  // TODO: Add parameters and argument labels
+
+  public var selectorName: String {
```

Comment:
Let's add a code comment to this

Reply:
7566b0f

Reply:
realizing I actually don't need this property at this level 🤦‍♂️. we can delete or leave it though. It might be useful to folk